### PR TITLE
[C++17] Check whether the output directory exists.

### LINKED ISSUE #10281: [C++17] step-70: Create output directory if necessary
See this comment: 
https://github.com/dealii/dealii/pull/10274/files#diff-5e19932b9e815b576e4bb4f4bacac9ca

There may be other steps to which this applies as well.

### PR DESCRIPTION
**Pull request by bangerth**
Fixes #10281 . We can only apply this once we support C++17, but I had this ready so I'll post for whenever the time comes.

### CODE REVIEW

File: examples/step-70/step-70.cc [hunk 1]
```diff
@@ -144,6 +144,7 @@ namespace LA
 #endif
 
 #include <cmath>
+#include <filesystem>
```
Comment (tamiko) [inline examples/step-70/step-70.cc:147]: That'll be some fun - For `libstdc++` we will have to figure out a sane way to decide whether we have to link against `libstdc++fs` as well or not.

Comment (bangerth) [reply examples/step-70/step-70.cc:147]: Shouldn't `-std=c++17` imply that?

Comment (tamiko) [reply examples/step-70/step-70.cc:147]: @bangerth gcc version 9 onwards should just require `-std=c++17`, gcc version 8 and older require manual linking with `libstdc++fs`. At least if I recall correctly :smile_cat:

Comment (marcfehling) [reply examples/step-70/step-70.cc:147]: I remember that I had similar issues with gcc8. I solved this by substituting them with `boost` libraries:
```
#ifdef __cpp_lib_filesystem
#  include <filesystem>
namespace filesystem
{
  using namespace std::filesystem;
}
#else
#  include <boost/filesystem.hpp>
namespace filesystem
{
  using namespace boost::filesystem;
}
#endif
```
I'm not sure if the `__cpp_lib_filesystem` flag only exists for `gcc`.


We could move this code snippet into the `std_cxx17` namespace. What do you think?

Comment (bangerth) [reply examples/step-70/step-70.cc:147]: The problem is that we don't include the boost filesystem parts in the `bundled/` directory :-( There is also the issue that this is a tutorial and I really didn't want to introduce some complicated workaround in tutorials -- I'd rather just leave things as they are if we can't do it right....

**Comment by bangerth** [comment]
This is now open for merge :-)

**Review by tamiko — APPROVED** [review]

**Comment by tamiko** [comment]
The minimal version we test and thus meaningfully support now is gcc-9 so this can go in as is.

**Review by kronbichler — APPROVED** [review]

**Comment by masterleinad** [comment]
> The minimal version we test and thus meaningfully support now is gcc-9 so this can go in as is.

tester-ng still uses gcc-7.5, though, see https://cdash.dealii.org/viewSite.php?siteid=3&project=1&currenttime=1688518800.

**Pull request merged**